### ISSUE TITLE
Cannot build browser.legacy on 1.8.2

### ISSUE THREAD

**Issue by pihentagy** [meteor/meteor#10776]
linux mint 19.2

Expected at least a descriptive error message what went wrong.

```

=> Started proxy.                             
                                              
/home/pihentagy/.meteor/packages/meteor-tool/.1.8.2-rc.10.1p1jh8f.tzs5k++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/meteor-promise/promise_server.js:218
      throw error;
      ^
[object Object]
```

**Comment by benjamn** [comment]
Is this an existing application, or a newly created one? Did this work with Meteor 1.8.1? Can you try to track down where in the beta/RC release process it started failing? There have been 19 beta releases (`1.8.2-beta.0` through `1.8.2-beta.18`) and 11 RCs (`1.8.2-rc.0` through `1.8.2-beta.10`), so a binary search might be a good idea.

**Comment by pihentagy** [comment]
It is an existing huge application, which used meteor 1.8.1. It was written in typescript and used barbatus:typescript. I'd like to try out native typescript support in meteor 1.8.2.

What I did so far:
```
 meteor npm install typescript@3.7
 meteor remove barbatus:typescript
 meteor add adornis:typescript
 meteor npm i @babel/helpers
```

So I can run them with:

```bash
for i in {0..9}; do
  meteor update --release 1.8.2-rc.$i
  meteor # successful run will stuck here
done
```

Right?

**Comment by pihentagy** [comment]
All rc releases failed to start (same error).

Now investigating the betas.

**Comment by afrokick** [comment]
You should use official package `meteor add typescript`

**Comment by pihentagy** [comment]
Figured it also myself, that it was a mistake to add adornis:typescript

Added typescript package, but got the same error :-/

**Comment by afrokick** [comment]
try to update babel
```
 "@babel/core": "7.7.2",
 "@babel/runtime": "7.7.2",
```

**Comment by pihentagy** [comment]
My `package.json` is below, the error message is the same for rc.10

I also tried `1.9.2-beta.2`, but `meteor update` failed :(
```json
{
  "name": "app",
  "private": true,
  "scripts": {
    "start": "meteor run",
    "test": "jest"
  },
  "dependencies": {
    "7.5": "^1.0.0",
    "@babel/helpers": "^7.7.0",
    "@babel/runtime": "^7.7.2",
    "@types/file-saver": "1.3.0",
    "apollo-cache-inmemory": "1.4.2",
    "apollo-client": "2.4.12",
    "apollo-link": "1.2.6",
    "apollo-link-ddp": "^2.1.0",
    "bcrypt": "^1.0.1",
    "body-parser": "^1.17.1",
    "cellref": "3.0.0",
    "colors": "1.1.2",
    "csv-parse": "^2.5.0",
    "exceljs": "^1.15.0",
    "express": "^4.17.1",
    "faker": "^4.1.0",
    "file-saver": "1.3.8",
    "filesize": "^3.5.6",
    "fixed-data-table": "^0.6.4",
    "fold-to-ascii": "4.0.0",
    "graphlib": "^2.1.1",
    "graphql": "14.1.1",
    "graphql-iso-date": "3.6.1",
    "graphql-redis-subscriptions": "2.1.0",
    "graphql-subscriptions": "1.1.0",
    "graphql-tag": "2.10.1",
    "graphql-tools": "4.0.4",
    "graphql-type-json": "0.2.1",
    "heapdump": "^0.3.15",
    "history": "^4.10.1",
    "http-proxy": "1.16.2",
    "indexof": "0.0.1",
    "inline-cpp": "github:lorros/node-inline-cpp#08d7ac516ccce5361fa88ba36d5571a9a4b6e705",
    "ioredis": "^4.14.1",
    "isomorphic-fetch": "^2.2.1",
    "ldapjs": "1.0.1",
    "lodash.clonedeep": "^4.5.0",
    "lodash.debounce": "^4.0.8",
    "lodash.flatmap": "4.5.0",
    "lodash.get": "^4.4.2",
    "lodash.isequal": "4.5.0",
    "lodash.memoize": "^4.1.2",
    "lodash.set": "4.3.2",
    "lodash.throttle": "^4.1.1",
    "lodash.uniq": "^4.5.0",
    "meteor-build-client": "^0.4.1",
    "meteor-node-stubs": "~0.4.1",
    "moment": "2.18.1",
    "pg": "7.5.0",
    "pg-bricks": "^0.6.0",
    "pg-escape": "^0.2.0",
    "pretty-ms": "^2.1.0",
    "re-resizable": "^4.7.1",
    "react": "^15.5.4",
    "react-addons-css-transition-group": "^15.5.2",
    "react-addons-pure-render-mixin": "^15.5.2",
    "react-addons-shallow-compare": "^15.5.2",
    "react-apollo": "2.0.4",
    "react-autosuggest": "9.3.4",
    "react-bootstrap": "0.31.3",
    "react-bootstrap-switch": "^15.0.4",
    "react-bootstrap-table": "^3.3.6",
    "react-chartist": "^0.10.2",
    "react-css-collapse": "^3.6.1",
    "react-datetime": "2.10.3",
    "react-dom": "^15.5.4",
    "react-draggable": "3.0.3",
    "react-google-charts": "^2.0.0",
    "react-google-maps": "github:gergely-ujvari/react-google-maps#746e38c140efc9783f9385b95cd5b5198314583b",
    "react-helmet": "^5.0.3",
    "react-redux": "5.0.6",
    "react-select": "^1.0.0-rc.10",
    "react-selectize": "^2.1.0",
    "react-transition-group": "^1.1.2",
    "redux": "3.7.0",
    "redux-form": "^6.7.0",
    "redux-thunk": "^2.1.0",
    "reselect": "^3.0.0",
    "responsive-fixed-data-table": "^2.0.0",
    "sanitize-filename": "1.6.1",
    "sanitize-html-react": "1.13.0",
    "screenfull": "^3.3.2",
    "shortid": "^2.2.15",
    "source-map-support": "0.5.10",
    "sql": "^0.74.0",
    "urijs": "^1.19.2",
    "xml2js": "^0.4.22"
  },
  "devDependencies": {
    "@types/classnames": "^0.0.32",
    "@types/colors": "1.1.3",
    "@types/csv-parse": "1.1.11",
    "@types/deep-diff": "0.0.31",
    "@types/enzyme": "^2.7.0",
    "@types/exceljs": "^0.5.2",
    "@types/express": "^4.17.2",
    "@types/faker": "^4.1.7",
    "@types/fixed-data-table": "^0.6.31",
    "@types/googlemaps": "^3.38.0",
    "@types/graphql": "14.0.5",
    "@types/heapdump": "^0.3.1",
    "@types/ioredis": "^4.0.19",
    "@types/jest": "^16.0.9",
    "@types/ldapjs": "^1.0.0",
    "@types/lodash": "^4.14.147",
    "@types/lodash.clonedeep": "^4.5.0",
    "@types/lodash.debounce": "^4.0.0",
    "@types/lodash.flatmap": "4.5.3",
    "@types/lodash.get": "^4.4.1",
    "@types/lodash.isequal": "4.5.2",
    "@types/lodash.memoize": "^4.1.0",
    "@types/lodash.set": "4.3.3",
    "@types/lodash.throttle": "^4.1.0",
    "@types/lodash.uniq": "^4.5.1",
    "@types/markerclustererplus": "^2.1.29",
    "@types/node": "^10.17.5",
    "@types/react": "^15.6.28",
    "@types/react-autosuggest": "9.3.4",
    "@types/react-bootstrap-table": "^2.6.3",
    "@types/react-dom": "^15.5.11",
    "@types/react-helmet": "^5.0.14",
    "@types/react-redux": "^4.4.40",
    "@types/react-select": "^1.0.58",
    "@types/redux-form": "^6.6.2",
    "@types/sanitize-filename": "1.1.28",
    "@types/screenfull": "^3.3.1",
    "@types/shortid": "^0.0.28",
    "@types/sinon": "^1.16.34",
    "@types/xml2js": "^0.0.33",
    "apollo-codegen": "github:csillag/apollo-codegen#92d7f5be4e688645d26e2560a2636a9906a3b2d4",
    "deep-diff": "0.3.8",
    "enzyme": "2.7.0",
    "imacss": "1.0.0",
    "jest": "^18.0.0",
    "react-addons-test-utils": "^15.3.2",
    "tslint": "^4.4.2",
    "tslint-react": "^2.3.0",
    "typescript": "^3.7.2"
  },
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(ts|tsx)$",
    "transform": {
      "^.+\\.(ts|tsx)$": "<rootDir>/jestPreprocessor.js"
    }
  }
}

```

**Comment by afrokick** [comment]
`"7.5": "^1.0.0",` - what is it?)

Did you try to add babel/core and meteor-node-stubs?
```
"@babel/core": "7.7.2",
"meteor-node-stubs": "1.0.0",
```

**Comment by pihentagy** [comment]
No, but now I did:

Still the same error:
```
/home/pihentagy/.meteor/packages/meteor-tool/.1.8.2.1hdxt8.1muvn++os.linux.x86_64+web.browser+web.browser.legacy+web.cordova/mt-os.linux.x86_64/dev_bundle/lib/node_modules/meteor-promise/promise_server.js:218
      throw error;
      ^
[object Object]
Makefile:66: recipe for target 'dev' failed
```

<details><summary>full package.json here</summary>
```json
{
  "name": "app",
  "private": true,
  "scripts": {
    "start": "meteor run",
    "test": "jest"
  },
  "dependencies": {
    "@babel/runtime": "^7.7.2",
    "@types/file-saver": "1.3.0",
    "apollo-cache-inmemory": "1.4.2",
    "apollo-client": "2.4.12",
    "apollo-link": "1.2.6",
    "apollo-link-ddp": "^2.0.1",
    "bcrypt": "^1.0.1",
    "body-parser": "^1.17.1",
    "cellref": "3.0.0",
    "colors": "1.1.2",
    "csv-parse": "^2.5.0",
    "exceljs": "^1.2.0",
    "express": "^4.15.2",
    "faker": "^4.1.0",
    "file-saver": "1.3.8",
    "filesize": "^3.5.6",
    "fixed-data-table": "^0.6.4",
    "fold-to-ascii": "4.0.0",
    "graphlib": "^2.1.1",
    "graphql": "14.1.1",
    "graphql-iso-date": "3.6.1",
    "graphql-redis-subscriptions": "2.1.0",
    "graphql-subscriptions": "1.1.0",
    "graphql-tag": "2.10.1",
    "graphql-tools": "4.0.4",
    "graphql-type-json": "0.2.1",
    "heapdump": "^0.3.9",
    "history": "^4.5.1",
    "http-proxy": "1.16.2",
    "indexof": "0.0.1",
    "inline-cpp": "github:lorros/node-inline-cpp#08d7ac516ccce5361fa88ba36d5571a9a4b6e705",
    "ioredis": "^4.14.1",
    "isomorphic-fetch": "^2.2.1",
    "ldapjs": "1.0.1",
    "lodash.clonedeep": "^4.5.0",
    "lodash.debounce": "^4.0.8",
    "lodash.flatmap": "4.5.0",
    "lodash.get": "^4.4.2",
    "lodash.isequal": "4.5.0",
    "lodash.memoize": "^4.1.2",
    "lodash.set": "4.3.2",
    "lodash.throttle": "^4.1.1",
    "lodash.uniq": "^4.5.0",
    "meteor-build-client": "^0.4.1",
    "meteor-node-stubs": "^1.0.0",
    "moment": "2.18.1",
    "pg": "7.5.0",
    "pg-bricks": "^0.6.0",
    "pg-escape": "^0.2.0",
    "pretty-ms": "^2.1.0",
    "re-resizable": "^4.7.1",
    "react": "^15.5.4",
    "react-addons-css-transition-group": "^15.5.2",
    "react-addons-pure-render-mixin": "^15.5.2",
    "react-addons-shallow-compare": "^15.5.2",
    "react-apollo": "2.0.4",
    "react-autosuggest": "9.3.4",
    "react-bootstrap": "0.31.3",
    "react-bootstrap-switch": "^15.0.4",
    "react-bootstrap-table": "^3.3.6",
    "react-chartist": "^0.10.2",
    "react-css-collapse": "^3.0.2",
    "react-datetime": "2.10.3",
    "react-dom": "^15.5.4",
    "react-draggable": "3.0.3",
    "react-google-charts": "^2.0.0",
    "react-google-maps": "github:gergely-ujvari/react-google-maps#746e38c140efc9783f9385b95cd5b5198314583b",
    "react-helmet": "^5.0.3",
    "react-redux": "5.0.6",
    "react-select": "^1.0.0-rc.10",
    "react-selectize": "^2.1.0",
    "react-transition-group": "^1.1.2",
    "redux": "3.7.0",
    "redux-form": "^6.7.0",
    "redux-thunk": "^2.1.0",
    "reselect": "^3.0.0",
    "responsive-fixed-data-table": "^2.0.0",
    "sanitize-filename": "1.6.1",
    "sanitize-html-react": "1.13.0",
    "screenfull": "^3.3.2",
    "shortid": "^2.2.6",
    "source-map-support": "0.5.10",
    "sql": "^0.74.0",
    "urijs": "^1.18.4",
    "xml2js": "^0.4.17"
  },
  "devDependencies": {
    "@types/classnames": "^0.0.32",
    "@types/colors": "1.1.3",
    "@types/csv-parse": "1.1.11",
    "@types/deep-diff": "0.0.31",
    "@types/enzyme": "^2.7.0",
    "@types/exceljs": "^0.5.2",
    "@types/express": "^4.11.1",
    "@types/faker": "^4.1.0",
    "@types/fixed-data-table": "^0.6.31",
    "@types/googlemaps": "^3.0.0",
    "@types/graphql": "14.0.5",
    "@types/heapdump": "^0.3.1",
    "@types/ioredis": "^4.0.10",
    "@types/jest": "^16.0.3",
    "@types/ldapjs": "^1.0.0",
    "@types/lodash": "^4.14.50",
    "@types/lodash.clonedeep": "^4.5.0",
    "@types/lodash.debounce": "^4.0.0",
    "@types/lodash.flatmap": "4.5.3",
    "@types/lodash.get": "^4.4.1",
    "@types/lodash.isequal": "4.5.2",
    "@types/lodash.memoize": "^4.1.0",
    "@types/lodash.set": "4.3.3",
    "@types/lodash.throttle": "^4.1.0",
    "@types/lodash.uniq": "^4.5.1",
    "@types/markerclustererplus": "^2.1.29",
    "@types/node": "^10.1.4",
    "@types/react": "^15.0.22",
    "@types/react-autosuggest": "9.3.4",
    "@types/react-bootstrap-table": "^2.6.3",
    "@types/react-dom": "^15.5.0",
    "@types/react-helmet": "^5.0.1",
    "@types/react-redux": "^4.4.40",
    "@types/react-select": "^1.0.58",
    "@types/redux-form": "^6.6.2",
    "@types/sanitize-filename": "1.1.28",
    "@types/screenfull": "^3.3.1",
    "@types/shortid": "^0.0.28",
    "@types/sinon": "^1.16.34",
    "@types/xml2js": "^0.0.33",
    "apollo-codegen": "github:csillag/apollo-codegen#92d7f5be4e688645d26e2560a2636a9906a3b2d4",
    "deep-diff": "0.3.8",
    "enzyme": "2.7.0",
    "imacss": "1.0.0",
    "jest": "^18.0.0",
    "react-addons-test-utils": "^15.3.2",
    "tslint": "^4.4.2",
    "tslint-react": "^2.3.0",
    "typescript": "^3.7.2"
  },
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js"
    ],
    "testRegex": "(/__tests__/.*|(\\.|/)(test|spec))\\.(ts|tsx)$",
    "transform": {
      "^.+\\.(ts|tsx)$": "<rootDir>/jestPreprocessor.js"
    }
  }
}
```
</details>

**Comment by pihentagy** [comment]
Okay, upgraded the project to 1.9.0-beta2, and can see the error message on the same line.

The error itself was not meteor related.

But the fact, that the error message is not displayed here, just [object Object] is a real issue, even on 1.8.2 release.

**Comment by nbedevi** [comment]
I have it too, what the solution?

**Comment by pihentagy** [comment]
For me the solution was to temporarily upgrade to 1.9 beta just to read the error message!

**Comment by filipenevola** [comment]
As this error was not in Meteor I'm going to close this one. Also, we just released 1.9 then you should be able to see the proper trace in 1.9.
